### PR TITLE
Standardize test locators to data-test-id

### PR DESCRIPTION
**Pull request by poulch**
#2865

### CODE REVIEW

File: src/Sidebar/MenuItem.tsx [hunk 1]
```diff
@@ -237,8 +233,7 @@ export const MenuItem: React.FC<MenuItemProps> = ({
                       onClick={(event: React.MouseEvent) =>
                         handleClick(event, subMenuItem)
                       }
-                      data-test="submenu-item-label"
-                      data-test-id={subMenuItem.id}
```
Comment (karolm-saleor) [inline src/Sidebar/MenuItem.tsx:241]: @krzysztofzuraw   need your expertise how to deal with entries like this, why are we using both of them atm?

Comment (krzysztofzuraw) [reply src/Sidebar/MenuItem.tsx:241]: I propose we merge those two into `data-text-id="submenu-item-label-${subMenuItem.id}"` and then change selectors inside cypress dashboard configuration.

**Review by karolm-saleor — APPROVED** [review]
LGTM
@karola312 FIY this will probably impact cloud e2e tests

**Comment by karolm-saleor** [comment]
Related to
https://github.com/saleor/saleor-dashboard/pull/2931

**Review by krzysztofzuraw — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Pull request merged**